catalog/app/utils/tracking.js

```diff
@@ -0,0 +1,45 @@
+import { take, select } from 'redux-saga/effects';
+
+const loadMixpanel = (token) =>
+  import('mixpanel-browser')
+    .then((mp) => {
+      mp.init(token);
+      return mp;
+    });
+
+const consoleTracker = Promise.resolve({
+  // eslint-disable-next-line no-console
+  track: (evt, opts) => console.log(`track: ${evt}`, opts),
+});
+
+export const mkTracker = (token) => {
+  const tracker = token ? loadMixpanel(token) : consoleTracker;
+
+  return {
+    nav: (loc, user) => tracker.then((inst) =>
+      inst.track('WEB', {
+        type: 'navigation',
+        origin: window.location.origin,
+        location: `${loc.pathname}${loc.search}${loc.hash}`,
+        user,
+      })),
+  };
+};
+
+export default function* tracking({
+  selectUsername,
+  locationChangeAction,
+  routerStartAction,
+  token,
+}) {
+  const tracker = mkTracker(token);
+
+  let started = false;
+
+  while (true) {
```

Comment:
for some reason, this makes me nervous. isn't it better to use `yield take(LOCATION_CHANGE|ROUTER_START, nav)`...?

Reply:
^ We could do that in App/sagas.js

Reply:
> We could do that in App/sagas.js

@akarve `App/sagas.js` is reworked heavily in #518, so i was avoiding doing anything there for now.

> for some reason, this makes me nervous.

what exactly?

>  isn't it better to use yield take(LOCATION_CHANGE|ROUTER_START, nav)...?

you mean `takeEvery`? like, forking on every action instead of using a blocking loop?